### ISSUE TITLE
Sync pg_dist_node when nodes join/leave

### ISSUE THREAD

**Issue by marcocitus** [citusdata/citus#797]
We need to propagate pg_dist_node changes to workers in order to perform distributed queries from workers. New nodes also need to receive a metadata snapshot.

Longer term, we could consider getting rid of any pg_dist_node dependencies in the query planner.

**Comment by marcocitus** [comment]
A change I'd like to propose compared to citus-mx is to rename the `nodeactive` column to `syncmetadata` and rename `cluster_activate_node(nodename,nodeport)` to something like `sync_metadata_to_node(nodename, nodeport)`, with the intent of decoupling the concepts of metadata being synced to a node and nodes being active, which are coupled in MX.

If we couple metadata syncing with the concept of a node being active (present in pg_dist_node), then syncing all pg_dist_node changes to all nodes becomes a requirement. However, in a non-MX Citus cluster (all tables have repmodel 'c'), there is no reason to sync changes to pg_dist_node or other metadata tables and doing so could create an operational burden. For example, you would not be able to add a new node if any other node is down, since syncing would fail. This burden is acceptable for MX clusters, which assume nodes are eventually replaced by their secondary, but would make regular Citus clusters much harder to operate.

The main functionality of `cluster_activate_node(nodename,nodeport)` is actually to a) sync metadata to the specified node, b) ensure all future metadata changes are propagated to the specified node. Therefore, renaming it to `sync_metadata_to_node` seems reasonable.

Apart from the renames, only a few small changes to the existing code are required, mainly:

- In worker_transaction.c we should send only to nodes with the `syncmetadata` flag enabled, it might make sense to rename the file and functions to reflect this change.
- master_add_node should propagate the insert into `pg_dist_node` to all nodes with `syncMetadata` enabled, because we might put shards on this node without calling `sync_metadata_to_node`. 
- `sync_metadata_to_node` should set the `syncmetadata` flag to true (instead of setting `nodeactive`), but propagating this update to workers is optional and can be skipped for now, since no operations on workers use this flag.

An additional benefit of decoupling metadata syncing is that we could easily add the explicit concepts of shards-only (data) nodes or distributed tables-only (aggregator) nodes if we'd wish to do so in the future.

**Comment by marcocitus** [comment]
Some things to note:
- In 6.0 we added support for foreign keys. When we sync metadata to workers then we need to make sure that the tables are created before the foreign keys are added. Otherwise foreign key creation would error out.
- In 6.0 we added support for truncate through triggers. We need to explicitly add these triggers as part of the DDL commands that we send to the workers.
- We ran into an issue in MX with the schema being created by super user, which made the schema inaccessible to regular users. We should make sure that the CREATE SCHEMA command specifies an AUTHORIZATION: https://www.postgresql.org/docs/current/static/sql-createschema.html
